src/cameras/PerspectiveCamera.js

```diff
@@ -99,6 +108,59 @@ class PerspectiveCamera extends Camera {
 
 	}
 
+	/*
+	 * Provides the topRight and bottomLeft corners of the camera's 2D frustum slice at a given distance.
+	 * Results are copied into minTarget and maxTarget input vars.
+	 */
+	getBounds( distance, minTarget, maxTarget ) {
+
+		_tempV3.set( - 1, - 1, 0.5 ).applyMatrix4( this.projectionMatrixInverse );
+		_tempV3.multiplyScalar( distance / Math.abs( _tempV3.z ) );
+		minTarget.x = _tempV3.x;
+		minTarget.y = _tempV3.y;
+
+		_tempV3.set( 1, 1, 0.5 ).applyMatrix4( this.projectionMatrixInverse );
+		_tempV3.multiplyScalar( distance / Math.abs( _tempV3.z ) );
+		maxTarget.x = _tempV3.x;
+		maxTarget.y = _tempV3.y;
+
+	}
+
+	/*
+	 * Calculates the height/width of the camera's frustum at a given distance.
+	 * returns a Vector2 where x is width and y is height.
+	 */
+	frustumDimensions( distance ) {
+
+		this.getBounds( distance, _minTarget, _maxTarget );
+		_dimensions.x = _maxTarget.x - _minTarget.x;
+		_dimensions.y = _maxTarget.y - _minTarget.y;
+
+		return _dimensions.clone();
+
+	}
+
+	/*
+	 * Calculates Vector3s of the frustum's corners at a given distance from the camera.
+	 * Returns an object with topLeft, topRight, bottomRight, and bottomLeft properties. Each property is a Vector3.
+	 */
+	frustumCorners( distance ) {
```

Comment:
This method belongs in `examples/jsm/utils/CameraUtils.js` IMO.

Can you please explain how these vertices are useful? It seems application-specific to me.

Reply:
Of the methods added this one is noteworthy in that its the only one to actually account for the camera's current position/scale/rotation and returns immediately usable 3D Vectors. 

I think it makes a strong case to remain in the PerspectiveCamera module given that it's a helper method that is specific to this class and provides easy access to information about the current camera.

Think it's more intuitive for newcomer developers to leave the method here. Specifically practical for situations where you want to manipulate objects relative to a moving camera or an oddly placed static camera.

Reply:
Sorry, I am trying to keep an open mind, but based on your answer, I am not hearing anything compelling.

Reply:
Okay sounds good guys. Will port this over to cameraUtils and we can revisit a migration to core if there's emergent common demand/adoption.

Reply:
frustumCorners renamed to getFrustumCorners and moved to CameraUtils https://github.com/mrdoob/three.js/pull/27574/commits/ab65b6e6974d23f62aa6cbdfcd80d6f03440fa4c